classes/external/enqueue_merge_request.php

```diff
@@ -0,0 +1,120 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Version information
+ *
+ * @package     tool
+ * @subpackage  mergeusers
+ * @author      Nicola Vallinoto, Liguria Digitale
+ * @author      Jordi Pujol-Ahulló, SREd, Universitat Rovira i Virgili
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace tool_mergeusers\external;
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once("{$CFG->libdir}/externallib.php");
+defined('MOODLE_INTERNAL') || die();
+use \tool_mergeusers\merge_request;
+use external_function_parameters;
+use external_multiple_structure;
+use external_single_structure;
+use external_value;
+class enqueue_merge_request extends \external_api {
+    /**
+     * Webservice enqueue_merge_request parameters
+     *
+     * @return external_function_parameters
+     */
+    public static function execute_parameters(): external_function_parameters {
+            return new external_function_parameters(
+            [
+                'removeuserfield' => new external_value(PARAM_TEXT, 'Remove user field'),
+                'removeuservalue' => new external_value(PARAM_RAW, 'Remove user value'),
+                'keepuserfield' => new external_value(PARAM_TEXT, 'Keep user field'),
+                'keepuservalue' => new external_value(PARAM_RAW, 'Keep user value'),
+            ]
+        );
+    }
+    /**
+     * Return identifier of the enqueue merge request
+     *
+     * @param $removeuserid
+     * @param $removeuservalue
+     * @param $keepuserid
+     * @param $keepuservalue
+     */
+    public static function execute(string $removeuserfield,
+                                     string $removeuservalue,
+                                   string $keepuserfield,
+                                   string $keepuservalue): int {
+        // Validate all of the parameters.
+        $params = self::validate_parameters(self::execute_parameters(),
+                                            ['removeuserfield' => $removeuserfield,
+                                            'removeuservalue' => $removeuservalue,
+                                            'keepuserfield' => $keepuserfield,
+                                            'keepuservalue' => $keepuservalue
+                                            ]);
+        global $DB;
+        // Insert of the merge request into mdl_merge_request_queue Moodle table.
+        $usertoremove = $DB->get_records(merge_request::TABLE_USERS,
+                                        [$removeuserfield => $removeuservalue]);
+        if (count($usertoremove) == 0) {
+            throw new Exception(get_string('cannotfindusertoremove', 'tool_mergeusers'));
```

Comment:
Add the third parameter to the get_string(,,) function, like `(object)['userfield' => $removeuserfield, 'uservalue'  => $removeuservalue]`, and then, the string should expect the variable `$a` with attributes `$a->userfield` and `$a->uservalue` you can use to provide the context. It is important to know which error was produced when failing.

Reply:
Sorry, I'm not talking about get_records(), but for the:

`throw new Exception(get_string('cannotfindusertoremove', 'tool_mergeusers', (object)['userfield' => $removeuserfield, 'uservalue'  => $removeuservalue]));`

This way, in the string, you can show the context and saying with user was not found, and not just one user was not found. When looking for why a merge request was done or not, all these details are really useful in the logs.